File: ios/RCTMGL/RCTMGLMapViewManager.m
#import <React/RCTBridgeModule.h>
#import <React/RCTViewManager.h>

@class RCTMGLMapView;

@interface RCT_EXTERN_MODULE(RCTMGLMapViewManager, RCTViewManager)

RCT_EXTERN_METHOD(addEvent:(NSString *)name location:(NSString *)location date:(nonnull NSNumber *)date)

@end

File: ios/RCTMGL/RCTMGLMapViewManager.swift
@objc(RCTMGLMapViewManager)
class RCTMGLMapViewManager: RCTViewManager {

    @objc(addEvent:location:date:)
    func addEvent(_ name: String, location: String, date: NSNumber) -> Void {
       // Date is ready to use!
    }

    @objc
    override static func requiresMainQueueSetup() -> Bool {
        return true
    }
    
    func defaultFrame() -> CGRect {
        return UIScreen.main.bounds
    }
    
    override func view() -> UIView! {
        let result = RCTMGLMapView(frame: self.defaultFrame())
        return result
    }
}
